generators/entity/index.js

```diff
@@ -493,13 +493,27 @@ class EntityGenerator extends BaseBlueprintGenerator {
   _loading() {
     return {
       loadEntity() {
+        loadRequiredConfigIntoEntity(this.context, this.jhipsterConfig);
+        this.context.disableSearch =
+          !this.context.searchEngine && this.jhipsterConfig.applicationType === MICROSERVICE && this.context.entityExisted;
```

Comment:
```suggestion
```

In some cases the search engine option should be stored at entity level.
So gateways can pick it later.
And in some cases search engine should be forced to disabled at entity level.
See my proposed changes:

```
--- a/generators/entity/index.js
+++ b/generators/entity/index.js
@@ -38,6 +38,8 @@ const { prepareFieldForTemplates, fieldIsEnum } = require('../../utils/field');
 const { prepareRelationshipForTemplates } = require('../../utils/relationship');
 const { stringify } = require('../../utils');
 const { GATEWAY, MICROSERVICE } = require('../../jdl/jhipster/application-types');
+const { NO: CLIENT_FRAMEWORK_NO } = require('../../jdl/jhipster/client-framework-types');
+const { NO: SEARCH_ENGINE_NO } = require('../../jdl/jhipster/search-engine-types');
 const { CASSANDRA, COUCHBASE, MONGODB, NEO4J, ORACLE, SQL } = require('../../jdl/jhipster/database-types');
 const {
   GENERATOR_ENTITIES,
@@ -359,6 +361,28 @@ class EntityGenerator extends BaseBlueprintGenerator {
           throw new Error(validation);
         }

+        if (this.entityConfig.searchEngine === undefined) {
+          const backendOnlyMicroservice =
+            context.applicationType === MICROSERVICE && (!context.clientFramework || context.clientFramework === CLIENT_FRAMEWORK_NO);
+          if (backendOnlyMicroservice && context.searchEngine && context.searchEngine !== SEARCH_ENGINE_NO) {
+            // Propagate searchEngine to the entity, it's opt-in and should be propagated to gateway
+            this.entityConfig.searchEngine = true;
+          }
+        } else if (
+          (!context.searchEngine || context.searchEngine === SEARCH_ENGINE_NO) &&
+          (context.applicationType !== GATEWAY || !this.entityConfig.microserviceName)
+        ) {
+          if (this.entityConfig.searchEngine === SEARCH_ENGINE_NO) {
+            // Convert to boolean.
+            this.entityConfig.searchEngine = false;
+          }
+          // If search engine is disabled at application, and unless it's a gateway publishing a microservice entity.
+          if (this.entityConfig.searchEngine) {
+            this.warning('Search engine disabled at application, cannot be enabled at entity, disabling');
+            delete this.entityConfig.searchEngine;
+          }
+        }
+
         this.entityConfig.name = this.entityConfig.name || context.name;
         if (![SQL, MONGODB, COUCHBASE, NEO4J].includes(context.databaseType)) {
           this.entityConfig.pagination = NO_PAGINATION;
```

Reply:
Implemented with one change - prooposed solution breaks test case when microservice has search engine disabled,  but entity has it enabled

Reply:
@ko5tik it's a new review with more changes.

Reply:
It also breaks this unit test case:
```

  2) JHipster generator for entity
       regeneration from json file
         microservice
           with dto, service, pagination
             shall not generate search specific artifacts because elastic search is false on top level:

```

Change  @497  is to satisfy this -   if   top search ist disable on  service level,   it has to be honored

Reply:
Ok, found problem -  deleting from entityConfig is not enough.    It has to be set to false

Reply:
changes are implemennted